pgrx-sql-entity-graph/src/pg_extern/returning.rs

```diff
@@ -54,297 +54,278 @@ impl Returning {
-                            let used_ty = UsedType::new(syn::Type::Path(typepath.clone()))?;
-                            Ok(Returning::Type(used_ty))
-                        }
-                    }
-                    syn::Type::Reference(ty_ref) => {
-                        let used_ty = UsedType::new(syn::Type::Reference(ty_ref.clone()))?;
+                            other => {
+                                return Err(syn::Error::new(
+                                    other.span(),
+                                    &format!("Got unexpected path argument: {other:?}"),
+                                ))
+                            }
+                        };
+                        Ok(Returning::Iterated {
+                            tys: iterated_items,
+                            optional: saw_option_ident,
+                            result: saw_result_ident,
+                        })
+                    } else {
+                        let used_ty = UsedType::new(syn::Type::Path(typepath.clone()))?;
                         Ok(Returning::Type(used_ty))
                     }
-                    syn::Type::Macro(ref mut type_macro) => Self::parse_type_macro(type_macro),
-                    syn::Type::Paren(ref mut type_paren) => match &mut *type_paren.elem {
-                        syn::Type::Macro(ref mut type_macro) => Self::parse_type_macro(type_macro),
-                        other => {
-                            return Err(syn::Error::new(
-                                other.span(),
-                                &format!("Got unknown return type: {type_paren:?}"),
-                            ))
-                        }
-                    },
-                    other => {
-                        return Err(syn::Error::new(
-                            other.span(),
-                            &format!("Got unknown return type: {other:?}"),
-                        ))
-                    }
+                } else {
+                    let used_ty = UsedType::new(syn::Type::Path(typepath.clone()))?;
+                    Ok(Returning::Type(used_ty))
+                }
+            }
+            syn::Type::Reference(ty_ref) => {
+                let used_ty = UsedType::new(syn::Type::Reference(ty_ref.clone()))?;
+                Ok(Returning::Type(used_ty))
+            }
+            syn::Type::Macro(ref mut type_macro) => Self::parse_type_macro(type_macro),
+            syn::Type::Paren(ref mut type_paren) => match &mut *type_paren.elem {
+                syn::Type::Macro(ref mut type_macro) => Self::parse_type_macro(type_macro),
+                other => {
+                    return Err(syn::Error::new(
+                        other.span(),
+                        &format!("Got unknown return type (type_paren): {type_paren:?}"),
+                    ))
                 }
+            },
+            syn::Type::Group(tg) => return Self::match_type(&tg.elem),
```

Comment:
This is the actual code change here -- supporting this variant.

All the surrounding code was moved into a new function called `match_type()` and we recurse here.  So the big diff is really just whitespace due to less indentation